atc/atccmd/command.go

```diff
@@ -96,387 +99,252 @@ const algorithmLimitRows = 100
 }
 
-func (cmd *Migration) supportedDBVersion() error {
-	helper := migration.NewOpenHelper(
-		defaultDriverName,
-		cmd.Postgres.ConnectionString(),
-		cmd.lockFactory,
-		nil,
-		nil,
-	)
-
-	version, err := helper.SupportedVersion()
-	if err != nil {
-		return err
-	}
-
-	fmt.Println(version)
-	return nil
-}
-
-func (cmd *Migration) migrateDBToVersion() error {
-	version := cmd.MigrateDBToVersion
-
-	var newKey *encryption.Key
-	var oldKey *encryption.Key
-
-	if cmd.EncryptionKey.AEAD != nil {
-		newKey = encryption.NewKey(cmd.EncryptionKey.AEAD)
-	}
-	if cmd.OldEncryptionKey.AEAD != nil {
-		oldKey = encryption.NewKey(cmd.OldEncryptionKey.AEAD)
-	}
-
-	helper := migration.NewOpenHelper(
-		defaultDriverName,
-		cmd.Postgres.ConnectionString(),
-		cmd.lockFactory,
-		newKey,
-		oldKey,
-	)
-
-	err := helper.MigrateToVersion(version)
-	if err != nil {
-		return fmt.Errorf("could not migrate to version: %d Reason: %s", version, err.Error())
-	}
-
-	fmt.Println("Successfully migrated to version:", version)
-	return nil
-}
-
-func (cmd *Migration) rotateEncryptionKey() error {
-	var newKey *encryption.Key
-	var oldKey *encryption.Key
-
-	if cmd.EncryptionKey.AEAD != nil {
-		newKey = encryption.NewKey(cmd.EncryptionKey.AEAD)
-	}
-	if cmd.OldEncryptionKey.AEAD != nil {
-		oldKey = encryption.NewKey(cmd.OldEncryptionKey.AEAD)
-	}
-
-	helper := migration.NewOpenHelper(
-		defaultDriverName,
-		cmd.Postgres.ConnectionString(),
-		cmd.lockFactory,
-		newKey,
-		oldKey,
-	)
-
-	version, err := helper.CurrentVersion()
-	if err != nil {
-		return err
-	}
-
-	return helper.MigrateToVersion(version)
-}
-
-func (cmd *Migration) migrateToLatestVersion() error {
-	helper := migration.NewOpenHelper(
-		defaultDriverName,
-		cmd.Postgres.ConnectionString(),
-		cmd.lockFactory,
-		nil,
-		nil,
-	)
-
-	version, err := helper.SupportedVersion()
-	if err != nil {
-		return err
-	}
-
-	return helper.MigrateToVersion(version)
-}
-
-func (cmd *ATCCommand) WireDynamicFlags(commandFlags *flags.Command) {
-	cmd.RunCommand.WireDynamicFlags(commandFlags)
-}
-
-func (cmd *RunCommand) WireDynamicFlags(commandFlags *flags.Command) {
-	var (
-		metricsGroup      *flags.Group
-		policyChecksGroup *flags.Group
-		credsGroup        *flags.Group
-		authGroup         *flags.Group
-	)
-
-	groups := commandFlags.Groups()
-	for i := 0; i < len(groups); i++ {
-		group := groups[i]
-
-		if credsGroup == nil && group.ShortDescription == "Credential Management" {
-			credsGroup = group
-		}
-
-		if metricsGroup == nil && group.ShortDescription == "Metrics & Diagnostics" {
-			metricsGroup = group
-		}
-
-		if policyChecksGroup == nil && group.ShortDescription == "Policy Checking" {
-			policyChecksGroup = group
-		}
-
-		if authGroup == nil && group.ShortDescription == "Authentication" {
-			authGroup = group
-		}
-
-		if metricsGroup != nil && credsGroup != nil && authGroup != nil && policyChecksGroup != nil {
-			break
-		}
+var CmdDefaults RunConfig = RunConfig{
```

Comment:
I like it - I think this is a lot easier to skim!

Reply:
:D Really?

I was worried people might like seeing the default as a tag next to the field more